lib/Onyx.js

```diff
@@ -18,6 +20,12 @@ const METHOD = {
     CLEAR: 'clear',
 };
 
+// A map controlling concurrent storage modification.
+// Each key is associated with a promise of an ongoing storage operation.
+// This allows sequential chaining of overlapping writes to the same key,
+// effectively preventing concurrent modification conflicts.
```

Comment:
Wouldn't it technically be possible then, that the earlier `set` call wins, because e.g. the later one takes longer due to a lot of data being written?

I think we can't really control how long it really takes to write data to storage and without the queuing we cannot 100% assure, that the value from latest `set` call will be written in the end..

Reply:
Regardless of the duration it takes - in the internal (native) structure, write operations occur sequentially. Hence, it's improbable for a later `set` to complete before an earlier `set`. Instead, it would hold off until the prior `set` operation is done, a process that happens at a lower level. 

The queue we're constructing serves to facilitate the particular Onyx `merge` behavior which require access to the most recent data. However, reading the latest data isn't feasible if a `set` is concurrently modifying the data key in question.

Reply:
Formally requesting here that we write a clear explanation in the code as a comment so that anyone who will read this can understand why we have this queue and how it works. Providing an example would also help visualize things. The current comment requires more detail.